internal/providers/proxmoxve/proxmoxve.go

```diff
@@ -123,20 +125,38 @@ func fetchConfigFromDevice(logger *log.Logger, ctx context.Context, path string)
 		)
 	}()
 
-	if !fileExists(filepath.Join(mnt, cidataPath)) {
-		return nil, nil
-	}
+	paths := []string{ciuserdataPath, civendordataPath}
+	header := []byte("#cloud-config\n")
 
-	contents, err := os.ReadFile(filepath.Join(mnt, cidataPath))
-	if err != nil {
-		return nil, err
-	}
+	for _, path := range paths {
+			fullPath := filepath.Join(mnt, path)
+			if !fileExists(fullPath) {
+					continue
+			}
 
-	header := []byte("#cloud-config\n")
-	if bytes.HasPrefix(contents, header) {
-		logger.Debug("config drive (%q) contains a cloud-config configuration, ignoring", path)
-		return nil, nil
+			contents, err := os.ReadFile(fullPath)
+			if err != nil {
+					// Log the error but continue to next file
+					logger.Debug("failed to read %q: %v", fullPath, err)
+					continue
+			}
+
+			// Skip if it's a cloud-config file
+			if bytes.HasPrefix(contents, header) {
+					logger.Debug("config drive (%q) contains a cloud-config configuration, ignoring", fullPath)
+					continue
+				}
+
+				// Try to validate if it's JSON
+				if json.Valid(contents) {
```

Comment:
Why check for JSON here? I prefer the current behaviour of "if it's not cloud-config data, assume that it's an Ignition config and otherwise fail". Though I guess we need to at least check that it's a non-zero file.

Reply:
It was a simple way to check that the file isn't empty and that it contains at least some valid JSON.  I think it is fine to leave as is.

Reply:
Hmm, again would rather we be more strict here. We should either expect cloud-init configs, Ignition configs, or an empty file, nothing else.

As it is for example, a typo in the user's Ignition config will be ignored rather than hard fail the system. But also I think given the messiness of this platform, I'd rather we always know exactly what we're dealing with as a way to keep track of any changes in expectations.

Reply:
In the current implementation we don't check if it is valid JSON or valid Ignition config. It only checks the first line and if it contains `#cloud-config` and if so it ignores the file. This means it won't ignore the file if they have invalid Ignition and hard fail the system. Checking for JSON at least ensures that there isn't an empty file but I can just revert it back to the current implmentation if you want.

There is only so much that can be done to protect against invalid Ignition configuration and I felt at least checking for valid JSON data is better than nothing.

Reply:
I just pushed an update that implements checking to ensure the file contains data before returning and removes the check for valid JSON data.